prExplorer/client-app/src/features/products/dashboard/ProductList.tsx

```diff
@@ -1,44 +1,139 @@
+  );
 
   return (
     <Segment>
-      <Item.Group divided>
-        {products.map((product) => (
-          <Item key={product.id}>
-            <Item.Content>
-              <Item.Header as="a">{product.title}</Item.Header>
-              <Item.Meta>Price: {product.price} - Discount : {product.discountPercentage} - Rating: {product.rating}</Item.Meta>
-              <Item.Description>
-                <div>{product.description}</div>
-                <div>STOCK: {product.stock}</div>
-              </Item.Description>
-              <Item.Extra>
-                <Button onClick={() => selectProduct(product.id)} floated="right" content='view' color='blue'/>
-                <Button name = {product.id} loading={submitting && target === product.id} onClick={(e) => handleProductDelete(e, product.id)} floated="right" content='delete' color='red'/>
-                <Label basic content={product.category} />
-              </Item.Extra>
-            </Item.Content>
-          </Item>
-        ))}
-      </Item.Group>
+      <Input
+        placeholder="Search..."
+        value={searchTerm}
+        onChange={(e) => setSearchTerm(e.target.value)}
+        action={{
+          icon: "search",
+          onClick: () => setSearchTerm(""),
+        }}
+      />
+      <Table sortable celled>
+        <Table.Header>
+          <Table.Row>
+            <Table.HeaderCell
+              sorted={column === "title" ? direction : undefined}
+              onClick={handleSort("title")}
+            >
+              Name
+            </Table.HeaderCell>
+            <Table.HeaderCell
+              sorted={column === "price" ? direction : undefined}
+              onClick={handleSort("price")}
+            >
+              Price
+            </Table.HeaderCell>
+            <Table.HeaderCell
+              sorted={column === "discountPercentage" ? direction : undefined}
+              onClick={handleSort("discountPercentage")}
+            >
+              Discount
+            </Table.HeaderCell>
+            <Table.HeaderCell
+              sorted={column === "rating" ? direction : undefined}
+              onClick={handleSort("rating")}
+            >
+              Rating
+            </Table.HeaderCell>
+            <Table.HeaderCell
+              sorted={column === "stock" ? direction : undefined}
+              onClick={handleSort("stock")}
+            >
+              Stock
+            </Table.HeaderCell>
+            <Table.HeaderCell
+              sorted={column === "category" ? direction : undefined}
+              onClick={handleSort("category")}
+            >
+              Category
+            </Table.HeaderCell>
+            <Table.HeaderCell>Thumbnail</Table.HeaderCell> {/* Yeni sütun */}
+            <Table.HeaderCell>View</Table.HeaderCell>
+            <Table.HeaderCell>Delete</Table.HeaderCell>
+          </Table.Row>
+        </Table.Header>
+        <Table.Body>
+          {filteredProducts.map((product) => (
```

Comment:
That implementation of sorting/filtering does the job. Nicely done!
However, for the future you can take a look at [react-hook-form](https://www.react-hook-form.com/) or [AG grid](https://www.ag-grid.com/react-data-grid/), which provide most of these features out of the box